lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -69,13 +68,104 @@ def _flatten(iterable):
+        first_iteration_output = next(generator, empty_generator_sentinel)
+        generator_is_empty = first_iteration_output == empty_generator_sentinel
+
+        if generator_is_empty:
+            yield default
+
+        else:
+            yield first_iteration_output
+            for element in generator:
+                yield element
+
+    def _batch_users(self, context):
+        """
+        Returns a generator of batches of users.
+        """
+
+        def grouper(iterable, chunk_size=100, fillvalue=None):
+            args = [iter(iterable)] * chunk_size
+            return zip_longest(*args, fillvalue=fillvalue)
+
+        def get_enrolled_learners_for_course(course_id, verified_only=False):
+            """
+            Get all the enrolled users in a course chunk by chunk.
+
+            This generator method fetches & loads the enrolled user objects on demand which in chunk
+            size defined. This method is a workaround to avoid out-of-memory errors.
+            """
+            filter_kwargs = {
+                'courseenrollment__course_id': course_id,
+            }
+            if verified_only:
+                filter_kwargs['courseenrollment__mode'] = CourseMode.VERIFIED
+
+            user_ids_list = get_user_model().objects.filter(**filter_kwargs).values_list('id', flat=True).order_by('id')
+            user_chunks = grouper(user_ids_list)
+            for user_ids in user_chunks:
+                user_ids = [user_id for user_id in user_ids if user_id is not None]
+                min_id = min(user_ids)
+                max_id = max(user_ids)
+                users = get_user_model().objects.filter(
+                    id__gte=min_id,
+                    id__lte=max_id,
+                    **filter_kwargs
+                ).select_related('profile')
+                yield users
+
+        course_id = context.course_id
+
+        report_for_verified_only = generate_grade_report_for_verified_only()
+        return get_enrolled_learners_for_course(course_id=course_id, verified_only=report_for_verified_only)
+
+    def _compile(self, context, batched_rows):
+        """
+        Compiles and returns the complete list of (success_rows, error_rows) for
+        the given batched_rows and context.
+        """
+        # partition and chain successes and errors
+        success_rows, error_rows = zip(*batched_rows)
+        success_rows = list(chain(*success_rows))
+        error_rows = list(chain(*error_rows))
+
+        # update metrics on task status
+        context.task_progress.succeeded = len(success_rows)
+        context.task_progress.failed = len(error_rows)
+        context.task_progress.attempted = context.task_progress.succeeded + context.task_progress.failed
+        context.task_progress.total = context.task_progress.attempted
+        return success_rows, error_rows
+
+    def _upload(self, context, success_headers, success_rows, error_headers, error_rows, report_name):
```

Comment:
As discussed, can we change this to have only `success_rows` and we combine the header and actual rows in the parent method? This will help in reducing the args being passed to this method.